tensorflow/core/kernels/data/cache_dataset_ops.cc

```diff
@@ -12,128 +12,258 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-#include "tensorflow/core/framework/dataset.h"
+#include "tensorflow/core/kernels/data/cache_dataset_ops.h"
+
 #include "tensorflow/core/framework/partial_tensor_shape.h"
 #include "tensorflow/core/framework/resource_mgr.h"
 #include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/kernels/data/name_utils.h"
 #include "tensorflow/core/lib/core/errors.h"
 #include "tensorflow/core/lib/strings/stringprintf.h"
 #include "tensorflow/core/platform/env.h"
 #include "tensorflow/core/util/tensor_bundle/tensor_bundle.h"
 
 namespace tensorflow {
 namespace data {
-namespace {
 
 // See documentation in ../../ops/dataset_ops.cc for a high-level description of
 // the following op.
 
-class CacheDatasetOp : public UnaryDatasetOpKernel {
+/* static */ constexpr const char* const CacheDatasetOp::kDatasetType;
+/* static */ constexpr const char* const CacheDatasetOp::kInputDataset;
+/* static */ constexpr const char* const CacheDatasetOp::kFileName;
+/* static */ constexpr const char* const CacheDatasetOp::kOutputTypes;
+/* static */ constexpr const char* const CacheDatasetOp::kOutputShapes;
+
+constexpr char kTensorStrFormat[] = "%%%zuzu_%%%zuzu";
```

Comment:
Is this variable name `kTensorStrFormat` OK? Would `kKeyStrFormat` be better

Reply:
`kKeyStrFormat` works for me